tools/check_release.py

```diff
@@ -0,0 +1,455 @@
+#  },
+#  "test_list" : [
+#    {
+#        "name" : "test_compile_mbed_lib",
+#        "lib" : "mbed"
+#    },
+#    {
+#        "name" : "test_compile_mbed_dev",
+#        "lib" : "mbed-dev"
+#    }
+#  ]
+#}
+#
+# The mbed_repo_path field should be changed to point to where your local
+# working directory is for Mercurial repositories.
+# For each test app you wish to run, add an entry to the test list. The example
+# above has 2 test apps  
+#     "test_compile_mbed_lib" and "test_compile_mbed_dev"
+# The lib field in each says which type of mbed 2 library the app contains.
+# These test apps MUST be available as repos in the user's online Mercurial area.
+#
+# Run the script from the mbed-os directory as follows:
+# > python tools/check_release.py 
+#
+# It will look for local clones of the test app repos. If they don't exist
+# it will clone them. It will then read the latest versions of mbed and mbed-dev
+# (an assumption is made that both of these are already cloned in your Mercurial area).
+# The lib files within the test apps are then updated to the corresponding version in 
+# the associated lib itself. The test apps are then committed and pushed back to the users
+# fork.
+# The test apps will then be compiled for all supported targets and a % result output at the 
+# end.
+#
+# Uses the online compiler API at https://mbed.org/handbook/Compile-API
+# Based on the example from https://mbed.org/teams/mbed/code/mbed-API-helper/
+ 
+ 
+import os, getpass, sys, json, time, requests, logging
+from os.path import dirname, abspath, basename, join
+import argparse
+import subprocess
+import re
+import hglib
+
+# Be sure that the tools directory is in the search path
+ROOT = abspath(join(dirname(__file__), ".."))
+sys.path.insert(0, ROOT)
+
+from tools.build_api import get_mbed_official_release
+
+OFFICIAL_MBED_LIBRARY_BUILD = get_mbed_official_release('2')
+
+def log_message(str):
+    """ Depending on the value of the global variable, dbg, writes a log message provided in 
+        str to stdout.
+        
+    Args:
+    str - log string.
+
+    """
+    global dbg
+    if dbg:
+        print str
+        sys.stdout.flush()
+
+
+def get_compilation_failure(messages):
+    """ Reads the json formatted 'messages' and checks for compilation errors.
+        If there is a genuine compilation error then there should be a new message containing
+        a severity field = Error and an accompanying message with the compile error text.
+        Any other combination is considered an internal compile engine failure 
+    Args:
+    messages - json formatted text returned by the online compiler API.
+    
+    Returns:
+    Either a string containing a compilation error or "Internal" to indicate an error with
+    the online IDE API itself.
+
+    """
+    for m in messages:
+        if 'severity' in m and 'message' in m:
+            if m['severity'] == 'error':
+                return m['message']
```

Comment:
That should be fine for our purposes. We only really want to know that the compilation failed and what the first compilation error was.